src/web_interface/rest/rest_firmware.py

```diff
@@ -106,7 +78,54 @@ def _get_parameters_from_request(request_parameters):
             raise ValueError('Inverted flag can only be used with recursive')
         return query, recursive, inverted, offset, limit
 
-    def _get_with_uid(self, uid):
+    @roles_accepted(*PRIVILEGES['submit_analysis'])
+    @api.expect(firmware_model)
+    def put(self):
+        '''
+        Update existing firmware analysis
+        You can use this endpoint to update a firmware analysis which is already existing
+        '''
+        try:
+            data = convert_rest_request(request.data)
+        except TypeError as type_error:
+            return error_message(str(type_error), self.URL, request_data=request.data)
+
+        result = self._process_data(data)
+        if 'error_message' in result:
+            logging.warning('Submission not according to API guidelines! (data could not be parsed)')
+            return error_message(result['error_message'], self.URL, request_data=data)
+
+        logging.debug('Upload Successful!')
+        return success_message(result, self.URL, request_data=data)
+
+    def _process_data(self, data):
+        for field in ['device_name', 'device_class', 'device_part', 'file_name', 'version', 'vendor', 'release_date',
+                      'requested_analysis_systems', 'binary']:
+            if field not in data.keys():
+                return dict(error_message='{} not found'.format(field))
+
+        data['binary'] = standard_b64decode(data['binary'])
+        firmware_object = convert_analysis_task_to_fw_obj(data)
+        with ConnectTo(InterComFrontEndBinding, self.config) as intercom:
+            intercom.add_analysis_task(firmware_object)
+        data.pop('binary')
+
+        return dict(uid=firmware_object.uid)
+
+
+@api.route('/<string:uid>',
+           doc={'description': '',
+                'params': {'uid': 'File UID'}
+                }
+           )
+class RestFirmwareGetWithUid(RestFirmwareBase):
+    @roles_accepted(*PRIVILEGES['view_analysis'])
+    @api.doc(responses={200: 'Success', 400: 'Unknown file object'})
+    def get(self, uid):
+        '''
+        Request a specific file
+        Get a specific file by providing the uid of the corresponding firmware
```

Comment:
This is kind of misleading. You don't really get a "file" in the result. It is a firmware entry from the database (converted to JSON).

Reply:
This is also missing the documentation of an optional URL parameter: "summary"